src/actions/getDomainMetadata.ts

```diff
@@ -0,0 +1,21 @@
+import { Registrar } from "../contracts/types";
+import { DomainMetadata } from "../types"
+import { makeApiCall } from "../api/actions/helpers";
+
+export const getDomainMetadata = async (domainId: string, registrar: Registrar) => {
+  const metadataUri = await registrar.tokenURI(domainId);
+
+  // DomainURI is always one of the two below
+  // https://ipfs.fleek.co/ipfs/Qm...
+  // ipfs://Qm...
```

Comment:
`ipfs://` can't be fetched via an http request (http:// vs ipfs://) which is what `makeApiCall` does. You will need to convert the `ipfs://QMHash` to http using an http IPFS gateway (like you're doing with `https://ipfs.io/ipfs/QMHash`)